### DISCUSSION TITLE
Non Parametric models training

### DISCUSSION THREAD

**Discussion by kaiserliche** [microsoft/onnxruntime#16212] [Other Q&A]
Can non parametric models like decision trees be trained on-device using onnxruntime ? I am trying to add support for training tree based models, but can't find a way to do it. Please help.

**Comment by Craigacp** [comment]
I think the training support is purely gradient based. There are other packages which should be able to train trees & tree ensembles on android devices, but I don't really know anything about iOS so it depends what platform you want.

**Reply by kaiserliche** [reply]
Yeah that's what I thought. Can you mention some packages that can train these kind of models on android devices ? Thanks for the reply 🙂

**Reply by Craigacp** [reply]
Well, I work on one called [Tribuo](https://github.com/oracle/tribuo) which has a tree ensemble implementation in pure Java that should work fine on Android. I've not tested it on that platform though, we're focused more on server applications, but if you hit issues you can raise them on our GitHub. I suspect [SMILE](https://github.com/haifengl/smile) would work just fine on Android too, but again I've not tested that.
